src/Schema/Directives/UploadDirective.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Nuwave\Lighthouse\Schema\Directives;
+
+use Exception;
+use Illuminate\Http\UploadedFile;
+use Nuwave\Lighthouse\Support\Contracts\ArgDirective;
+use Nuwave\Lighthouse\Support\Contracts\ArgTransformerDirective;
+
+class UploadDirective extends BaseDirective implements ArgDirective, ArgTransformerDirective
+{
+    public static function definition(): string
+    {
+        return /** @lang GraphQL */ <<<'GRAPHQL'
+"""
+Uploads given file to storage, removes the argument and sets 
+the returned path to the attribute key provided.
+
+This does not change the schema from a client perspective.
+"""
+directive @upload(
+  """
+  The storage disk to be used, defaults to config value `filesystems.default`.
+  """
+  disk: String
+  """
+  The path where the file should be stored.
+  """
+  path: String! = "/"
+) on ARGUMENT_DEFINITION | INPUT_FIELD_DEFINITION
+GRAPHQL;
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function transform($argumentValue): ?string
+    {
+        if ($argumentValue === null) {
+            return null;
+        }
+
+        if (!($argumentValue instanceof UploadedFile)) {
+            throw new Exception("Expected UploadedFile from `{$this->name()}`");
+        }
+
+        $filename = $argumentValue->hashName();
+
+        $filepathInStorage = $argumentValue->storeAs(
+            $this->pathArgValue(),
+            $filename,
+            $this->diskArgValue()
+        );
+
+        if ($filepathInStorage === false) {
+            throw new Exception("Unable to upload `{$this->name()}` file to `{$this->pathArgValue()}` via disk `{$this->diskArgValue()}`");
```

Comment:
Can we use a more appropriate exception class - maybe something from the framework? Alternatively, this would be a `RuntimeException`.

Also `$this->name()` is the name of the directive, which does not really help much here. Let's include the name of the argument (`$this->nodeName()`).